File: src/main/java/dao/FileDao.java
package dao;

import models.File;
import org.hibernate.NonUniqueResultException;
import org.hibernate.Session;
import org.hibernate.Transaction;
import utils.FileDBSessionFactoryUtil;

import javax.persistence.NoResultException;
import javax.persistence.criteria.CriteriaQuery;
import javax.persistence.criteria.Root;
import java.util.List;

public class FileDao extends Dao<File> {

    private int userID;
    private String password;

    public FileDao(int userID, String password) {
        this.userID = userID;
        this.password = password;
    }

    public File findById(int id) {
        return FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession().get(File.class, id);
    }

    public File findByNamePath(String name, String path) {
        Session session = FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession();
        CriteriaQuery<File> criteriaQuery = session.getCriteriaBuilder().createQuery(File.class);
        Root<File> root = criteriaQuery.from(File.class);
        criteriaQuery.multiselect(root.get("name"), root.get("path"));
        criteriaQuery.where(
                session.getCriteriaBuilder().equal(root.get("name"), name),
                session.getCriteriaBuilder().equal(root.get("path"), path)
        );

        File result = null;
        try {
            result = session.createQuery(criteriaQuery).getSingleResult();
        } catch (NonUniqueResultException e) {
            e.printStackTrace();
        } catch (NoResultException ignored) {}

        return result;
    }

    public void save(File file) {
        Session session = FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession();
        Transaction tx = session.beginTransaction();
        session.save(file);
        tx.commit();
        session.close();
    }

    public void update(File file) {
        Session session = FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession();
        Transaction tx = session.beginTransaction();
        session.update(file);
        tx.commit();
        session.close();
    }

    public void delete(File file) {
        Session session = FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession();
        Transaction tx = session.beginTransaction();
        session.delete(file);
        tx.commit();
        session.close();
    }

    public List<File> getAll() {
        return (List<File>) FileDBSessionFactoryUtil.getSessionFactory(userID, password).openSession().createQuery("from File").list();
    }
}

File: src/main/kotlin/dao/FileDao.kt
@file:Suppress("SyntaxError")

package dao

import models.File
import org.hibernate.NonUniqueResultException
import utils.SessionFactoryUtil
import javax.persistence.NoResultException

class FileDao(private val userID: Int, private val password: String) : Dao<File>() {

    override val all: List<File>
        get() = SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()?.createQuery("from File")?.list() as List<File>

    override fun findById(id: Int): File? {
        return SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()?.get(File::class.java, id)
    }

    fun findByNamePath(name: String, path: String): File? {
        val session = SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()
        val criteriaQuery = session?.criteriaBuilder?.createQuery(File::class.java)
        val root = criteriaQuery?.from(File::class.java)
        criteriaQuery?.multiselect(root?.get<File>("id"), root?.get<File>("name"), root?.get<File>("path"), root?.get<File>("content"))
        criteriaQuery?.where(
                session.criteriaBuilder?.equal(root?.get<File>("name"), name),
                session.criteriaBuilder?.equal(root?.get<File>("path"), path)
        )

        var result: File? = null
        try {
            result = session?.createQuery(criteriaQuery)?.singleResult
        } catch (e: NonUniqueResultException) {
            e.printStackTrace()
        } catch (ignored: NoResultException) {
        } finally {
            session.close()
        }

        return result
    }

    override fun save(obj: File) {
        val session = SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()
        val tx = session?.beginTransaction()
        session?.save(obj)
        tx?.commit()
        session?.close()
    }

    override fun update(obj: File) {
        val session = SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()
        val tx = session?.beginTransaction()
        session?.update(obj)
        tx?.commit()
        session?.close()
    }

    override fun delete(obj: File) {
        val session = SessionFactoryUtil.getSessionFactory(userID, password)!!.openSession()
        val tx = session?.beginTransaction()
        session?.delete(obj)
        tx?.commit()
        session?.close()
    }

}
